mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoDeleteMethodGenerator.java

```diff
@@ -187,7 +204,7 @@ public DaoDeleteMethodGenerator(
           entityDefinition.getPrimaryKey().stream()
               .map(PropertyDefinition::getCqlName)
               .collect(Collectors.toList());
-      List<? extends VariableElement> bindMarkers = parameters.subList(0, primaryKeyNames.size());
+      List<? extends VariableElement> bindMarkers = parameters.subList(0, keyParameterSize);
```

Comment:
I think this assertion in `GeneratedCodePatterns.bindParameters` is going to fail with this change:
```java
assert bindMarkerNames.size() == parameters.size();
```
I think the cleanest fix is to sublist `primaryKeyNames` as well.

Reply:
good catch :+1: